Scripts/Mac/GenerateCommonAssemblyVersion.sh

```diff
@@ -1,11 +1,16 @@
 . "$(dirname ${BASH_SOURCE[0]})/InitializeEnvironment.sh"
 
-GVFSPROPS=$VFS_SRCDIR/GVFS/GVFS.Build/GVFS.props
-VERSIONNUMBER="$(cat $GVFSPROPS | grep GVFSVersion | grep -Eo '[0-9.]+(-\w+)?')"
+if [ -z $1 ]; then
+  echo "Version Number not defined for CommonAssemblyVersion.cs"
+fi
 
+# Update the version number in GVFS.props for other consumers of GVFSVersion
+sed -i "" -E "s@<GVFSVersion>[0-9]+(\.[0-9]+)*</GVFSVersion>@<GVFSVersion>$1</GVFSVersion>@g" $VFS_SRCDIR/GVFS/GVFS.Build/GVFS.props
```

Comment:
Do we really need the regex for the version number?  Isn't it hard coded in the GVFS.props?  Should we just hard code it here too?

Reply:
I used a regex rather than hardcoding the build number for scenarios where the user runs a build with a custom build number multiple times in a row. If we hardcode the number, this scenario wouldn't work (we wouldn't match the GVFSVersion line once it's set to 1.0.0.0): 
```
./BuildGVFSForMac.sh Debug 1.0.0.0
./BuildGVFSForMac.sh Debug 1.0.0.1
```

I doubt anyone will ever set the version outside of CI/CD so maybe this is a moot point. I could clarify in the comment since Regex's tend to be hard to reason about.